README.md

```diff
@@ -2,68 +2,81 @@
+Shadertoys translated to wgsl are also supported using the uniforms `i_resolution`, `i_time`, etc. 
 
-This code is distributed under the 2-clause BSD license.
+This project is not affiliated with Shadertoy.com
 
-## Developers
+## Installation
+```bash
+pip install git+https://github.com/pygfx/shadertoy
+```
 
-* Clone the repo.
-* Create a virtual environment using `python -m venv .venv`
-* Install using `.venv/bin/pip install -e .[dev]`
-* Use `.venv/bin/ruff format .` to apply autoformatting.
-* Use `.venv/bin/ruff --fix .` to run the linter.
-* Use `.venv/bin/pytest .` to run the tests.
-* Use `.venv/bin/pip wheel -w dist --no-deps .` to build a wheel.
+## Usage
 
-*Note*: Replace `/bin/` with `/Scripts/` on Windows.
+The main `Shadertoy` class takes shader code as a string.
 
-## Testing
+```python
+from wgpu_shadertoy import Shadertoy
 
-The test suite is divided into multiple parts:
+shader_code = """
+void mainImage( out vec4 fragColor, in vec2 fragCoord )
+{
+    vec2 uv = fragCoord/iResolution.xy;
+    vec3 col = 0.5 + 0.5*cos(iTime+uv.xyx+vec3(0,2,4));
+    fragColor = vec4(col,1.0);
+}
+"""
 
-* `pytest -v tests` runs the core unit tests.
-* `pytest -v examples` tests the examples.
+shader = Shadertoy(shader_code, resolution=(800, 450))
 
-There are two types of tests for examples included:
+if __name__ == "__main__":
+    shader.show()
+```
 
-### Type 1: Checking if examples can run
+Texture inputs are supported by using the `ShadertoyChannel` class. Up to 4 channels are supported.
 
-When running the test suite, pytest will run every example in a subprocess, to
-see if it can run and exit cleanly. You can opt out of this mechanism by
-including the comment `# run_example = false` in the module.
+```python
+from wgpu_shadertoy import Shadertoy, ShadertoyChannel
+from PIL import Image
+import numpy as np
 
-### Type 2: Checking if examples output an image
+shader_code = """
+void mainImage( out vec4 fragColor, in vec2 fragCoord )
+{
+    vec2 uv = fragCoord/iResolution.xy;
+    vec4 c0 = texture(iChannel0, 2.0*uv + iTime * 0.2);
+    fragColor = c0;
+}
+"""
 
-You can also (independently) opt-in to output testing for examples, by including
-the comment `# test_example = true` in the module. Output testing means the test
-suite will attempt to import the `canvas` instance global from your example, and
-call it to see if an image is produced.
+image_data = np.array(Image.open("./examples/screenshots/shadertoy_star.png"))
+channel0 = ShadertoyChannel(image_data, wrap="repeat")
+shader = Shadertoy(shader_code, resolution=(800, 450), inputs=[channel0])
+```
 
-To support this type of testing, ensure the following requirements are met:
+When passing `off_screen=True` the `.snapshot()` method allows you to render specific frames.
+```python
+shader = Shadertoy(shader_code, resolution=(800, 450), off_screen=True)
+frame0_data = shader.snapshot()
+frame10_data = shader.snapshot(10.0)
```

Comment:
I would recommend to expand this example a little to show how to write a snapshot to a .png file